File: 0874-backspace-string-compare/0874-backspace-string-compare.java
class Solution {
    public boolean backspaceCompare(String s, String t) {
        
        Stack<Character> st = new Stack<>();
        Stack<Character> st1 = new Stack<>();
        int i =0, j=0;
        while(i != s.length()){ 

            char a  = s.charAt(i);
            if(a == '#'){
            if(!st.isEmpty()){
                st.pop();
             
            }
           
             
            }else{
                st.push(a);
           
            }
            i++;
        }
         while(j !=t.length()){
             char b  = t.charAt(j);

             if(b == '#'){
            if(!st1.isEmpty()){
                st1.pop();

            }
         
            }else{
                st1.push(b);
              
            }
            j++;
         }


        
        String a ="",b = "";
        while(!st.isEmpty()){
            a = a+ st.pop();
            
           
        }
          while(!st1.isEmpty()){
            b =b+ st1.pop();
           
            
        }

        if(a.equals(b)) return true;
        else return false;
    }
}

File: 0874-backspace-string-compare/0874-backspace-string-compare.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        stack<char>st1; stack<char>st2;
        int i=0;
        while(s[i] != '\0'){
            if(s[i] == '#'){
               if(!st1.empty()) st1.pop();
            }
            else st1.push(s[i]);
            i++;
        }
        i=0;
        while(t[i] != '\0'){
            if(t[i]  == '#'){
                if(!st2.empty()) st2.pop();
            }else st2.push(t[i]);
            i++;
        }
        if(st1.empty() && st2.empty()) return true;

        while(!st1.empty() && !st2.empty()){
            if(st1.top() != st2.top()){ return false;}

            st1.pop(); st2.pop();
        }
        if((st1.empty() && !st2.empty()) || (!st1.empty() && st2.empty())) return false;
        return true;

    }
};